### PR TITLE
Added test for transactional JDBC sink

### PR DESCRIPTION
**Pull request by olukas**

### CODE REVIEW

File: snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    private int sleepMsBetweenItem;
+    private long snapshotIntervalMs;
+
+    public static void main(String[] args) throws Exception {
+        new JdbcSinkTest().run(args);
+    }
+
+    @Override
+    public void init(JetInstance client) throws Exception {
+        connectionUrl = System.getProperty("connectionUrl", DEFAULT_DATABASE_URL)
```
Comment (gurbuzali) [inline snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkTest.java:63]: `property(String, String)` method from `AbstractSoakTest` can be used here

File: snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+    }
+
+    private Pipeline pipeline(String tableName) {
+        int sleep = sleepMsBetweenItem;
+
+        Pipeline pipeline = Pipeline.create();
+
+        StreamSource<Integer> source = SourceBuilder
```
Comment (gurbuzali) [inline snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkTest.java:122]: we use this source for other tests too, time to move it to a central place I guess. can be done as a separate PR

File: snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkVerifier.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    private final String name;
+    private final ILogger logger;
+    private final String connectionUrl;
+
+    private volatile boolean finished;
+    private volatile Throwable error;
+    private long counter;
+    private final PriorityQueue<Integer> verificationQueue = new PriorityQueue<>();
+
+    public JdbcSinkVerifier(String name, ILogger logger, String connectionUrl) {
```
Comment (gurbuzali) [inline snapshot-jdbc-test/src/main/java/com/hazelcast/jet/tests/snapshot/jdbc/JdbcSinkVerifier.java:50]: I also see some similarities between JmsSinkVerifier, GeneratedFilesVerifier and JdbcSinkVerifier. some of the logic can be centralized

**Comment by olukas** [comment]
Thanks for review. I created https://github.com/hazelcast/hazelcast-jet-ansible-tests/issues/79 to track review comments related to logic centralization.

**Pull request merged**